kernel/timeout.c

```diff
@@ -310,6 +310,7 @@ uint64_t z_timeout_end_calc(k_timeout_t timeout)
 		return UINT64_MAX;
 	} else if (K_TIMEOUT_EQ(timeout, K_NO_WAIT)) {
 		return z_tick_get();
+	} else {
```

Comment:
If the previous if clauses were all early-exits like they are here, you can put the remainder of the block into the final else as a clearer way to conform to the rule.